sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java

```diff
@@ -2873,6 +2870,149 @@ private void formatRawResultDate(Object[] result, int index) {
 		}
 	}
 
+	private void migrateVaccinationInfo() throws SQLException {
+		// Retrieve all new and unsynchronized cases with vaccinationStatus == VACCINATED from the database
+		GenericRawResults<Object[]> caseInfoResult = getDao(Case.class).queryRaw(
+			"SELECT cases.id, person_id, disease, diseaseDetails, reportDate, reportingUser_id, responsibleRegion_id, responsibleDistrict_id, "
+				+ "responsibleCommunity_id, cases.creationDate, firstVaccinationDate, vaccinationDate, vaccinationDoses, vaccineName, otherVaccineName, vaccine"
+				+ "vaccineManufacturer, otherVaccineManufacturer, vaccinationInfoSource, vaccineInn, vaccineBatchNumber, vaccineUniiCode, vaccineAtcCode,"
+				+ "pregnant, trimester, healthConditions_id FROM cases LEFT JOIN clinicalCourse ON cases.clinicalCourse_id = clinicalcourse.id WHERE snapshot = 0 AND changeDate = 0 AND vaccination = 'VACCINATED';",
+			new DataType[] {
+				DataType.BIG_INTEGER,
+				DataType.BIG_INTEGER,
+				DataType.ENUM_STRING,
+				DataType.STRING,
+				DataType.DATE_LONG,
+				DataType.BIG_INTEGER,
+				DataType.BIG_INTEGER,
+				DataType.BIG_INTEGER,
+				DataType.BIG_INTEGER,
+				DataType.DATE_LONG,
+				DataType.DATE_LONG,
+				DataType.DATE_LONG,
+				DataType.STRING,
+				DataType.ENUM_STRING,
+				DataType.STRING,
+				DataType.STRING,
+				DataType.ENUM_STRING,
+				DataType.STRING,
+				DataType.ENUM_STRING,
+				DataType.STRING,
+				DataType.STRING,
+				DataType.STRING,
+				DataType.STRING,
+				DataType.ENUM_STRING,
+				DataType.ENUM_STRING,
+				DataType.BIG_INTEGER });
+
+		for (Object[] caseInfo : caseInfoResult) {
+			formatRawResultString(caseInfo, 2, false);
+			formatRawResultString(caseInfo, 3, true);
+			formatRawResultString(caseInfo, 12, true);
+			formatRawResultString(caseInfo, 14, true);
+			formatRawResultString(caseInfo, 15, true);
+			formatRawResultString(caseInfo, 16, false);
+			formatRawResultString(caseInfo, 17, true);
+			formatRawResultString(caseInfo, 18, false);
+			formatRawResultString(caseInfo, 19, true);
+			formatRawResultString(caseInfo, 20, true);
+			formatRawResultString(caseInfo, 21, true);
+			formatRawResultString(caseInfo, 22, true);
+			formatRawResultString(caseInfo, 23, false);
+			formatRawResultString(caseInfo, 24, false);
+			formatRawResultDate(caseInfo, 4);
+			formatRawResultDate(caseInfo, 9);
+			formatRawResultDate(caseInfo, 10);
+			formatRawResultDate(caseInfo, 11);
+		}
+
+		// Retrieve the latest case of each person for each disease
```

Comment:
Why only the latest? Don't we have to do this for all cases of each person?

Reply:
No, because we only want one immunization per person/disease combination